server/issue.go

```diff
@@ -867,49 +858,121 @@ func (s *Server) createDatabaseCreateTaskList(ctx context.Context, c api.CreateD
 	}, nil
 }
 
-// creates PITR TaskCreate list and dependency.
-func createPITRTaskList(database *api.Database, projectID int, targetTs int64) ([]api.TaskCreate, []api.TaskIndexDAG, error) {
+func (s *Server) createPITRTaskList(ctx context.Context, originDatabase *api.Database, projectID int, c api.PITRContext) ([]api.TaskCreate, []api.TaskIndexDAG, error) {
 	var taskCreateList []api.TaskCreate
+	var taskIndexDAGList []api.TaskIndexDAG
 
-	// task: create and restore to PITR database
+	// Restore payload
 	payloadRestore := api.TaskDatabasePITRRestorePayload{
-		ProjectID:     projectID,
-		PointInTimeTs: &targetTs,
+		ProjectID: projectID,
+	}
+
+	if c.CreateDatabaseCtx != nil {
+		targetInstance, err := s.store.GetInstanceByID(ctx, c.CreateDatabaseCtx.InstanceID)
+		if err != nil {
+			return nil, nil, fmt.Errorf("failed to find the instance with ID: %d, err: %w", c.CreateDatabaseCtx.InstanceID, err)
+		}
+		project, err := s.store.GetProjectByID(ctx, projectID)
+		if err != nil {
+			return nil, nil, fmt.Errorf("failed to find the project with ID: %d, err: %w", projectID, err)
+		}
+		taskList, err := s.createDatabaseCreateTaskList(ctx, *c.CreateDatabaseCtx, *targetInstance, *project)
+		if err != nil {
+			return nil, nil, fmt.Errorf("failed to create the database create task list, error: %w", err)
+		}
+		taskCreateList = append(taskCreateList, taskList...)
+
+		payloadRestore.TargetInstanceID = &targetInstance.ID
+		payloadRestore.DatabaseName = &c.CreateDatabaseCtx.DatabaseName
 	}
+	if c.BackupID != nil {
+		// We only support restore the full backup to a new database only, so c.CreateDatabaseCtx must be not nil in this case.
+		if c.CreateDatabaseCtx == nil {
+			return nil, nil, common.Errorf(common.Invalid, "expect craete a new database in restore backup %d issue, but get nil", *c.BackupID)
+		}
+
+		backup, err := s.store.GetBackupByID(ctx, *c.BackupID)
+		if err != nil {
+			return nil, nil, fmt.Errorf("failed to find backup %v", c.BackupID)
+		}
+		if backup == nil {
+			return nil, nil, fmt.Errorf("backup not found with ID %d", c.BackupID)
+		}
+
+		if err != nil {
+			return nil, nil, fmt.Errorf("failed to create restore database task, unable to marshal payload %w", err)
+		}
+
+		payloadRestore.BackupID = c.BackupID
+
+		bytesRestore, err := json.Marshal(payloadRestore)
+		if err != nil {
+			return nil, nil, fmt.Errorf("failed to create PITR restore task, unable to marshal payload, error: %w", err)
+		}
+
+		taskIndexDAGList = append(taskIndexDAGList, api.TaskIndexDAG{
+			FromIndex: len(taskCreateList) - 1,
```

Comment:
Is it possible to avoid using "len(taskCreateList) - 1"?

Reply:
We can hard core it likes [FromIndex: 0, ToIndex: 1].
The merge method is part of TaskDAG part. 
Let's hard code here and refactor it after completing a common function to merge it. WDYT?

Reply:
Hard-coding is the best actually I think.

Reply:
Done.